financial/simple_moving_average.py

```diff
@@ -0,0 +1,59 @@
+"""
+Calculate the Simple Moving Average (SMA) for a time series data.
+https://en.wikipedia.org/wiki/Moving_average
+"""
+
+
+def simple_moving_average(data: list[int], window_size: int) -> list[float | None]:
+    """
+
+
+    :param data: A list of numerical data points.
+    :param window_size: An integer representing the size of the SMA window.
+    :return: A list of SMA values with the same length as the input data.
+
+    The Simple Moving Average (SMA) is a statistical calculation used to
+    analyze data points by creating
+    a constantly updated average price over a specific time period.
+    In finance, SMA is often used in technical
+    analysis to smooth out price data and identify trends.
```

Comment:
```suggestion
    Calculate the simple moving average (SMA) for some given time series data.

    :param data: A list of numerical data points.
    :param window_size: An integer representing the size of the SMA window.
    :return: A list of SMA values with the same length as the input data.
```

Let's keep the simple function explanation in the docstring and have the more in-depth explanation of the algorithm at the top of the file.